yarn.config.cjs

```diff
@@ -0,0 +1,184 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/** @type {import('@yarnpkg/types')} */
+const { defineConfig } = require('@yarnpkg/types');
+const fg = require('fast-glob');
+
+/**
+ * The workspace object used in the constraints function
+ * @typedef {import('@yarnpkg/types').Yarn.Constraints.Workspace} Workspace
+ */
+
+module.exports = defineConfig({
+    async constraints({ Yarn }) {
+        /**
+         * Fetch a list of all the component workspaces using a glob pattern
+         * @type {string[]} components
+         */
+        const components = fg.sync('packages/*', {
+            cwd: __dirname,
+            onlyDirectories: true,
+        });
+
+        /**
+         * This function checks the workspace for any local package references
+         * and ensure that the devDependencies are up-to-date with the latest version
+         * currently in the project
+         * @param {Workspace} workspace
+         * @returns {void}
+         */
+        function validateLocalPackages(workspace) {
+            // Return early if the workspace does not have any peerDependencies
+            if (!workspace.manifest.peerDependencies) {
+                return;
+            }
+
+            // Start by filtering out the local packages from the external dependencies
+            const localPackages = Object.keys(
+                workspace.manifest.peerDependencies
+            )
+                .filter((pkg) => Yarn.workspace({ ident: pkg }))
+                .map((pkg) => Yarn.workspace({ ident: pkg }));
+
+            // Iterate over the local packages and ensure that the devDependencies are up-to-date
+            for (const localWorkspace of localPackages) {
+                const localVersion = localWorkspace.manifest.version;
+                workspace.set(
+                    `devDependencies.${localWorkspace.manifest.name}`,
+                    localVersion ?? 'workspace:^'
+                );
+            }
+        }
+
+        /**
+         * A reusable function to add keywords to ensure workspaces
+         * include a minimal set of keywords for discoverability
+         * with additionalKeywords as an optional parameter to add more
+         * specific keywords that are relevant to the workspace
+         * @param {string[]} additionalKeywords
+         * @returns {string[]}
+         */
+        function keywords(additionalKeywords = []) {
+            return [
+                'design-system',
+                'spectrum',
+                'adobe',
+                'adobe-spectrum',
+                'web components',
+                'web-components',
+                'lit-element',
+                'lit-html',
```

Comment:
Can we add the values to a new array like `baseKeyWords` and use this? Saves re-allocation of new array everytime keywords is called!

Reply:
Minimally beneficial and I'm not sure the abstraction is worth it. If the script was slow to run, I'd say the optimization makes sense but I'm not sure it's worth more abstraction in an already foreign syntax to most contributors.